string user_name
string type_train 
---